db/SDC_postgreSQL/dataGenerator.js

```diff
@@ -0,0 +1,124 @@
+const faker = require('faker');
+
+// get a random number between min(inclusive) and max(non-inclusive).
+// [decimalPlaces]: optional. default to 0(integer) if not provided
+const getRandomNum = (min, max, decimalPlaces) => {
+  decimalPlaces = decimalPlaces || 0;
+  const multiplier = Math.pow(10, decimalPlaces);
+  const minAdj = min * multiplier;
+  const maxAdj = ((max - 1) * multiplier) + 1;
+
+  const randomNumAdj = Math.floor(Math.random() * (maxAdj - minAdj)) + minAdj;
+  return randomNumAdj / multiplier;
+};
+
+// pick a random item from an array
+const pickRandomItem = array => array[getRandomNum(0, array.length)];
+
+const settings = {
+  similarListingsPerId: 10,
+
+  id: {
+    min: 0,
+    max: 200, // non-inclusive
+  },
+
+  title: {
+    minWords: 2,
+    maxWords: 8, // non-inclusive
+  },
+
+  type: {
+    options: ['Entire Place', 'Shared Room', 'Private Room'],
+  },
+
+  numBeds: {
+    min: 1,
+    max: 20, // non-inclusive
+  },
+
+  price: {
+    min: 10,
+    max: 2000, // non-inclusive
+  },
+
+  numRatings: {
+    min: 0,
+    max: 2000, // non-inclusive
+  },
+
+  avgStars: {
+    min: 0,
+    max: 6, // non-inclusive
+    decimalPlaces: 1,
+  },
+
+  thumbnailImage: {
+    width: 316,
+    height: 210,
+  },
+};
+
+const generateOneSimilarListing = id => ({
+  id: id,
+
+  title: faker.lorem.words(getRandomNum(
+    settings.title.minWords,
+    settings.title.maxWords,
+  )),
+
+  type: pickRandomItem(settings.type.options,),
+
+  numbeds: getRandomNum(
+    settings.numBeds.min,
+    settings.numBeds.max,
+  ),
+
+  price: getRandomNum(
+    settings.price.min,
+    settings.price.max,
+  ),
+
+  numratings: getRandomNum(
+    settings.numRatings.min,
+    settings.numRatings.max,
+  ),
+
+  avgstars: getRandomNum(
+    settings.avgStars.min,
+    settings.avgStars.max,
+    settings.avgStars.decimalPlaces,
+  ),
+
+  thumbnailimage: 'https://picsum.photos/' +
+                    `${settings.thumbnailImage.width}/` +
+                    `${settings.thumbnailImage.height}` +
+                    `?image=${id}`,
+});
+
+const insertIntoDatabase = (collection, array) => collection.insertMany(array);
+
+const batchInsertSimilarListings = (batchNumber, collection) => {
```

Comment:
Are you using these functions with postgres? These seem like they're left over from a Mongo implementation. If so, you might want to consider keeping different implementations on different files or different branches

Reply:
good call, thanks!